### PR TITLE
♻️ State2 unit test replacement

### PR DESCRIPTION
**Pull request by tongzhi1998**
- Replace State with State2 for all unit tests (Below TP)

### CODE REVIEW

File: src/monad/execution/test/evmc_host.cpp [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
 using namespace execution;
 
-using traits_t = fake::traits::alpha<fake::State::ChangeSet>;
+struct fakeBlockCache
+{
+    [[nodiscard]] bytes32_t get_block_hash(uint64_t) const noexcept
+    {
+        return bytes32_t{};
+    }
+} block_cache;
```
Comment (sheapwilliams) [inline src/monad/execution/test/evmc_host.cpp]: We should push this into fakes.hpp....it is repeated *everywhere*.

File: src/monad/state2/test/state.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ TYPED_TEST_SUITE(StateTest, DBTypes);
 
 struct fakeBlockCache
 {
-    [[nodiscard]] bytes32_t get_block_hash(int64_t) const noexcept
+    [[nodiscard]] bytes32_t get_block_hash(uint64_t) const noexcept
```
Comment (sheapwilliams) [inline src/monad/state2/test/state.cpp]: ....with differing signatures...

File: src/monad/execution/test/evmc_host.cpp (hunk 1 above)
Comment (tongzhi1998) [reply src/monad/execution/test/evmc_host.cpp]: Used `fake::BlockDb` to replace the `fakeBlockCache` everywhere

File: src/monad/execution/test/block_processor.cpp [hunk 3]
```diff
@@ -35,11 +35,14 @@ using code_state_t = state::CodeState<db_t>;
 using real_state_t = state::State<
     account_state_t, value_state_t, code_state_t, block_db_t, db_t>;
 
+/*
+// TODO: Commented out due to change in fork_traits, and to avoid merge conflict
```
Comment (zander-xyz) [inline src/monad/execution/test/block_processor.cpp]: usually TODOs state what the todo is. so for example, "comment this back in when <xyz> is fixed"

File: src/monad/state2/test/state.cpp [hunk 4]
```diff
@@ -3,8 +3,13 @@
 
 #include <monad/db/in_memory_trie_db.hpp>
 #include <monad/db/rocks_trie_db.hpp>
+
+#include <monad/execution/test/fakes.hpp>
+
 #include <monad/state/state_changes.hpp>
```
Comment (zander-xyz) [inline src/monad/state2/test/state.cpp]: no longer needed

File: src/monad/execution/ethereum/test/fork_traits.cpp [hunk 5]
```diff
@@ -27,7 +25,12 @@ using namespace monad::fork_traits;
 using namespace monad::execution;
 
 using db_t = db::InMemoryTrieDB;
-using state_t = execution::fake::State::ChangeSet;
+using mutex_t = std::shared_mutex;
+using block_cache_t = execution::fake::BlockDb;
+
+using state_t = state::State<mutex_t, block_cache_t>;
+
+block_cache_t block_cache;
```
Comment (zander-xyz) [inline src/monad/execution/ethereum/test/fork_traits.cpp]: put this one below in the anonymous namespace

File: src/monad/state2/test/state.cpp (hunk 4 above)
Comment (tongzhi1998) [reply src/monad/state2/test/state.cpp]: Good catch!

**Pull request merged**